runtime/lua/vim/filetype/detect.lua

```diff
@@ -219,6 +219,24 @@ function M.cls(_, bufnr)
   return 'st'
 end
 
+--- *.cmd is close to a Batch file, but on OS/2 Rexx files and TI linker command files also use *.cmd.
+--- lnk: `/* comment */`, `// comment`, and `--linker-option=value`
+--- rexx: `/* comment */`, `-- comment`
+--- @type vim.filetype.mapfn
+function M.cmd(_, bufnr)
+  local lines = table.concat(getlines(bufnr, 1, 20))
+  if matchregex(lines, [[MEMORY\|SECTIONS\|\%(^\|\n\)--\S\|\%(^\|\n\)//]]) then
```

Comment:
This could probably be converted to `findany` Lua patterns, but I don't understand the last two regex.